### PR TITLE
feat(purchases): link stock entry from purchase

### LINKED ISSUE #4274: Link stock entry movement from purchase order when it has been received in stock
When a purchase order has been fulfilled, the purchase registry updates its to show "Received in Stock".  See below:

image
_Fig 1: Received in Stock Purchase Orders_

We should add an option to the dropdown menu to "Show Entry Movement" or something of the sort.  This would link the stock movements registry and show all the stock items that were received and tied to that purchase order.

### PR DESCRIPTION
**Pull request by jniles**
Adds a feature to link the stock entry movement from the purchase registry.  Additionally, I've implemented the `bhFilterToggle` on the purchase registry.

Closes #4274.

### CODE REVIEW

File: client/src/modules/purchases/list/list.js [hunk 1]
```diff
@@ -135,6 +143,11 @@ function PurchaseListController(
 
     PurchaseOrder.read(null, filters)
       .then((purchases) => {
+
+        purchases.forEach(purchase => {
+          purchase.hasStockMovement = !allowEditStatus(purchase.status_id);
+        });
```
Comment (mbayopanda) [inline client/src/modules/purchases/list/list.js:149]: 👍

**Review by mbayopanda — APPROVED** [review]
LGTM 👍 

bors r+

**Comment by mbayopanda** [comment]
bors try

**Comment by mbayopanda** [comment]
bors try

**Comment by jniles** [comment]
bors try

**Pull request merged**